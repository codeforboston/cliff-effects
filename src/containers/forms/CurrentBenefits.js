import { connect } from 'react-redux';

import { CurrentBenefitsStep } from '../../forms/CurrentBenefits';
import { setHasBenefit } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),

    USState: state.getIn([
      'geography',
      'state',
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setHasBenefit({ benefit, value }) {
      dispatch(
        setHasBenefit({
          time: 'current',
          benefit,
          value,
        })
      );
    },
  };
};

const CurrentBenefitsContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentBenefitsStep);

export { CurrentBenefitsContainer as CurrentBenefitsStep };
