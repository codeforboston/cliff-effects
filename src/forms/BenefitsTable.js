// REACT COMPONENTS
import React from 'react';
import {
  Table
} from 'semantic-ui-react';

// CUSTOM COMPONENTS
// Both the table and graph should just be added to a results page, but
// this will do for now

// BENEFIT LOGIC
import { getSNAPBenefits } from '../programs/federal/snap';
import { getHousingBenefit } from '../programs/state/massachusetts/housing';

// OBJECT MANIPULATION
import { cloneDeep } from 'lodash';


const getSignSymbol = function ( num ) {
  if ( num > 0 ) { return '+'; }
  else if ( num < 0 ) { return '-'; }
  else { return ''; }
};  // End getSignSymbol()


const BenefitsTable = function ( props ) {

  var currentClient = cloneDeep( props.client ),
      futureClient  = cloneDeep( currentClient );
  // Hack because everything's using future value instead
  // of just using current value.
  currentClient.future.earned = currentClient.current.earned;




  var SNAPBenefitCurrent  = Math.round( getSNAPBenefits( currentClient ).benefitValue * 12 ),
      SNAPBenefitFuture   = Math.round( getSNAPBenefits( futureClient ).benefitValue * 12 ),
      SNAPDiff            = SNAPBenefitFuture - SNAPBenefitCurrent,
      sec8BenefitCurrent  = Math.round( getHousingBenefit( currentClient ).benefitValue * 12 ),
      sec8BenefitFuture   = Math.round( getHousingBenefit( futureClient ).benefitValue * 12 ),
      sec8Diff            = sec8BenefitFuture - sec8BenefitCurrent,
      totalBenefitCurrent = SNAPBenefitCurrent + sec8BenefitCurrent,
      totalBenefitFuture  = SNAPBenefitFuture + sec8BenefitFuture,
      totalDiff           = SNAPDiff + sec8Diff,
      incomeCurrent       = Math.round( currentClient.current.earned * 12 ),
      incomeFuture        = Math.round( futureClient.future.earned * 12 ),
      incomeDiff          = incomeFuture - incomeCurrent,
      netCurrent          = totalBenefitCurrent + incomeCurrent,
      netFuture           = totalBenefitFuture + incomeFuture,
      netDiff             = totalDiff + incomeDiff;


const   columnHeaderStyle = {
                            background: 'rgba(0, 181, 173, 1)',
                            color: 'white',
                            fontSize: '1.3em',
                            fontWeight: 900,
                            textAlign: 'center',
                            borderRadius: 'inherit',
                            letterSpacing: '0.02em',
                            }
      , totalsRowStyle    = {
                            borderTop: "2px solid rgba(0, 181, 173, 1)",
                            fontWeight: 700,
                            fontSize: '1.1em',
                            padingTop: '0.25em'
                              }
      , rowHeaderStyle    = {
                    fontSize: '1.1em',
                    fontWeight: 700,
                    textAlign: 'left'
                    }
      , totalsRowHeaderStyle = {
                    fontSize: '1.2em',
                    fontWeight: 700,
                    textAlign: 'left',
                    borderTop: "2px solid rgba(0, 181, 173, 1)",
                    padingTop: '0.25em'


      };






const SNAPBenefitRow = function( props ){
  if(!currentClient.current.hasSnap ) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>SNAP</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(SNAPDiff) } ${Math.abs(SNAPDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };

const Sec8BenefitRow  = function( props ){  
  if(!currentClient.current.hasHousing) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Section 8 Housing</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(sec8Diff) } ${Math.abs(sec8Diff)} / year</Table.Cell>
      </Table.Row>
    ) 
};

const TotalBenefitsRow = function( props ){
  if(!currentClient.current.hasSnap || !currentClient.current.hasHousing) return (null)
    return(
      <Table.Row>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Total Benefits</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(totalDiff) } ${Math.abs(totalDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const IncomeRow = function ( props ) { 
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Income</Table.Cell>
        <Table.Cell textAlign='right'>${incomeCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${incomeFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(incomeDiff) } ${Math.abs(incomeDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const TotalsRow = function ( props ) {
    return (
      <Table.Row style={{border: 'none'}}>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Net Total</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(netDiff) } ${Math.abs(netDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };


  return (
    <wrapper>
      <Table celled>
       <Table.Header>
          <Table.Row >
            <Table.Cell style={columnHeaderStyle} width={3}>Benefit</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Current Benefits</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>New Estimate</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Difference</Table.Cell>
          </Table.Row>
      </Table.Header>
        <Table.Body>
          <SNAPBenefitRow client={currentClient} />
          <Sec8BenefitRow client={currentClient} />
          <TotalBenefitsRow client={currentClient} />
          <IncomeRow />
          <TotalsRow />
        </Table.Body>
      </Table>
    </wrapper>
  )

};  // End BenefitsTable(<>)


export {
	BenefitsTable
};
