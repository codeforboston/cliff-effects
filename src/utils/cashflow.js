/** For all income and general cash flow getters and calculators
* 
* @todo Add checks for properties with messages for mistakes
* @todo Deal with expiration dates
*/

// LOGIC
import { sum, pick, values } from 'lodash';

// DATA
import {
  UNEARNED_INCOME_SOURCES,
  UNDER13_CARE_EXPENSES,
  OVER12_CARE_EXPENSES,
} from '../data/massachusetts/name-cores';

// ==================================
// DEDUCTIONS
// ==================================
/** 
 * Total MONTHLY dependent costs, including for those under and over
 *     the age of 13 (does not include child support paid out).
 *     Dependent = child or disabled adult
 * 
 * @function
 * @param {object} client - `current` or `future` property of client data
 * 
 * @returns {number} - Total dependent care expenses
 */
const getDependentCostsMonthly = function (client) {
  var props = UNDER13_CARE_EXPENSES.concat(OVER12_CARE_EXPENSES);
  return sumProps(client, props);
};  // End getDependentCostsMonthly()


// ==================================
// STRAIGHT UP INCOME
// ==================================

/** 
 * Gets sum of all unearned monthly income of given client.
 * 
 * @function
 * @param {object} client - `current` or `future` property of client data
 * 
 * @returns {number}
 */
const getGrossUnearnedIncomeMonthly = function (client) {
  return sumProps(client, UNEARNED_INCOME_SOURCES);
};  // End getGrossUnearnedIncomeMonthly()


/**
 * Total monthly earned and unearned income with no deductions or
 *     exclusions.
 * 
 * @function
 * @param {object} client - `current` or `future` property of client data
 * 
 * @returns {number} - Total earned and unearned monthly
 *     income with no deductions or exclusions.
 */
const getSimpleGrossIncomeMonthly = function (client) {
  var earned    = client.earned,
      unearned  = getGrossUnearnedIncomeMonthly(client);
  return earned + unearned;
};  // End getSimpleGrossIncomeMonthly()


// ==================================
// INCOME HELPERS
// ==================================

/** 
 * Returns the sum of the requested properties of of a given object
 * 
 * @function
 * @param {object} obj - Has the properties named in `props` with number values.
 * @param {array} props - The names of some properties in `obj` with number values.
 * 
 * @returns {number}
 */
const sumProps = function (obj, props) {
  return sum(values(pick(obj, props)));
};  // End sumProps()


export {
  getDependentCostsMonthly,
  getSimpleGrossIncomeMonthly,
  getGrossUnearnedIncomeMonthly,
  sumProps,
};
