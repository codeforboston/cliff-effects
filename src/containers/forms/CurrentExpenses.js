import { connect } from 'react-redux';

import { CurrentExpensesStep } from '../../forms/CurrentExpenses';

import { setCashValue, setHousingType, setPaysUtility, setGetsFuelAssistance } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setExpenseValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },

    setHousingType({ housingType }) {
      dispatch(
        setHousingType({
          time: 'current',
          housingType,
        })
      );
    },

    setPaysUtility({ utility, paysUtility }) {
      dispatch(
        setPaysUtility({
          time: 'current',
          utility,
          paysUtility,
        })
      );
    },

    setGetsFuelAssistance({ getsAssistance }) {
      dispatch(
        setGetsFuelAssistance({
          time: 'current',
          getsAssistance,
        })
      );
    },
  };
};

const CurrentExpensesStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentExpensesStep);

export { CurrentExpensesStepContainer as CurrentExpensesStep };
