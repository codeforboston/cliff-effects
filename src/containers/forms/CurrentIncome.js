import { connect } from 'react-redux';

import { CurrentIncomeStep } from '../../forms/CurrentIncome';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setIncomeValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },
  };
};

const CurrentIncomeStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentIncomeStep);

export { CurrentIncomeStepContainer as CurrentIncomeStep };
