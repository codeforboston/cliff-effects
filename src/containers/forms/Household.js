import { connect } from 'react-redux';

import { HouseholdStep } from '../../forms/Household';
import {
  addMember,
  removeMember,
  setMemberIsDisabled,
  setMemberRole,
  setMemberAge,
} from '../../actions';

const time = 'current';

const mapStateToProps = (state) => {
  return {
    household: state.getIn([
      'client',
      'current',
      'household', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setMemberAge({ index, age }) {
      dispatch(
        setMemberAge({
          time,
          index,
          age,
        })
      );
    },

    setMemberIsDisabled({ index, isDisabled }) {
      dispatch(
        setMemberIsDisabled({
          time,
          index,
          isDisabled,
        })
      );
    },

    setMemberRole({ index, role }) {
      dispatch(
        setMemberRole({
          time,
          index,
          role,
        })
      );
    },

    removeMember({ index }) {
      dispatch(
        removeMember({
          time,
          index,
        })
      );
    },

    addMember({ member }) {
      dispatch(
        addMember({
          time,
          member,
        })
      );
    },
  };
};

const HouseholdStepContainer = connect(mapStateToProps, mapDispatchToProps)(HouseholdStep);

export { HouseholdStepContainer as HouseholdStep };
