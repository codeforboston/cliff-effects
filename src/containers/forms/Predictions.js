import { connect } from 'react-redux';

import { PredictionsStep } from '../../forms/Predictions';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return { client: state.get('client') };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setPredictionValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'future',
          name,
          value,
        })
      );
    },
  };
};

const PredictionsStepContainer = connect(mapStateToProps, mapDispatchToProps)(PredictionsStep);

export { PredictionsStepContainer as PredictionsStep };
