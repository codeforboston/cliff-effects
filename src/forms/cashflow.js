// REACT COMPONENTS
import React, { Component } from 'react';
import { Form } from 'semantic-ui-react';

// PROJECT COMPONENTS
import { ManagedNumberField } from './inputs';
import { InvalidMessage } from './formHelpers';

// UTILITIES
import { toMonthlyAmount } from '../utils/math';
import { isNonNegNumber, hasOnlyNonNegNumberChars } from '../utils/validators';
import { toMoneyStr } from '../utils/prettifiers';


/** Contains cash flow inputs, their label, and any user feedback
 *
 * @function
 * @param {object} props
 * @property {object} props.children - The cash flow inputs
 * @property {object} props.label - What category these inputs are for
 * @property {object} props.validRow - Whether 'invalid' feedback needs to be shown
 * @property {object} props.message - Feedback to user about invalid input
 *
 * @returns Component
 */
const CashFlowContainer = function ({ children, label, validRow, message }) {
  return (
    <Form.Field
      inline
      className={ 'cashflow' }>
      { children }
      <div className={ 'cashflow-column cashflow-column-last-child' }>
        <label>{ label }</label>
      </div>
      <InvalidMessage
        validRow={ validRow }
        message={ message } />
    </Form.Field>
  );
};  // End <CashFlowContainer>

/** Maximum input value of yearly income allowed in the textbox,
 * monthly/daily will be scaled
 */
const maximum_value_yearly = 999999.99;

/** One row for cash flow inputs - weekly, monthly, yearly
 *
 * @param {object} props
 * @property {object} props.generic - Base name for the client property that
 *     needs to be updated (now the code has changed, this may be a misnomer)
 * @property {object} props.timeState - Client, either future values or current values
 * @property {object} props.updateClientValue - Updates client state
 * @property {object} props.children - Text for the row label
 * 
 */
/** @todo Find elegant way to combine CashFlowInputsRow and MonthlyCashFlowRow
      use `includes` array to include only certain columns perhaps. */
class CashFlowInputsRow extends Component {
  constructor(props) {
    super(props);
    this.state = { message: '' };
  }

  // Special store validator that handles maximums and sets error message
  cashFlowStoreValidator = (max) => {
    return (str) => {
      if (!isNonNegNumber(str)) {
        this.setState({ message: 'Invalid number format' });
        return false;
      }
      else if (parseFloat(str) > max) {
        this.setState({ message: ('The input number exceeds the maximum of $' + maximum_value_yearly + '/yr') });
        return false;
      }
      this.setState({ message: '' });
      return true;
    };
  };

  render() {
    var { generic, timeState, updateClientValue, children } = this.props;

    var updateClient = function (evnt, inputProps, data) {
      var monthly = toMonthlyAmount[ data.interval ](evnt, inputProps.value),
          obj     = { name: generic, value: monthly };
      updateClientValue(evnt, obj);
    };
  
    
  
    /** baseVal
     * Get the time ('future' or 'current') monthly value unless there is
     *     none, in which case, get the 'current' monthly cash flow value
     *     (to prefill future values with 'current' ones if needed).
     *
     * @var
     *
     * @todo Add some kind of UI indication when it's the same as the 'current'
     *     value. What if some of the row's values are the same and some are
     *     different?
     */
    var baseVal   = timeState[ generic ],
        baseProps = {
          name:             generic,
          className:        'cashflow-column',
          store:            updateClient,
          displayValidator: hasOnlyNonNegNumberChars,
          format:           toMoneyStr,
        };

    var cashFlowStoreValidator = this.cashFlowStoreValidator;
  
    return (
      <CashFlowContainer
        label={ children }
        validRow={ !this.state.message }
        message={ this.state.message }>
        <ManagedNumberField
          storeValidator={ cashFlowStoreValidator(maximum_value_yearly / 52) }
          { ...baseProps }
          value     = { baseVal / (4 + 1 / 3) }
          otherData = {{ interval: 'weekly' }} />
        <ManagedNumberField
          storeValidator={ cashFlowStoreValidator(maximum_value_yearly / 12) }
          { ...baseProps }
          value     = { baseVal }
          otherData = {{ interval: 'monthly' }} />
        <ManagedNumberField
          storeValidator={ cashFlowStoreValidator(maximum_value_yearly) }
          { ...baseProps }
          value     = { baseVal * 12 }
          otherData = {{ interval: 'yearly' }} />
      </CashFlowContainer>
    );
  }
}


const CashFlowDisplayRow = function ({ generic, value, timeState, children }) {

  var baseVal      = value || timeState[ generic ],
      colClassName = `cashflow-column`,
      weekly       = toMoneyStr(baseVal / (4 + 1 / 3)),
      monthly      = toMoneyStr(baseVal),
      yearly       = toMoneyStr(baseVal * 12);

  /** @todo Make label a link to the input row */
  return (
    <div className = { `cashflow cashflow-display` }>
      <div className = { colClassName + ` ` + generic + ` output-number` } >
        { weekly }
      </div>
      <div className = { colClassName + ` ` + generic + ` output-number` } >
        { monthly }
      </div>
      <div className = { colClassName + ` ` + generic + ` output-number` } >
        { yearly }
      </div>
      <div className = { colClassName + ` cashflow-column-last-child` }>
        <label>{ children }</label>
      </div>

    </div>
  );

};  // End <CashFlowDisplayRow>


/** One row for _one_ cash flow input - a monthly value
 *
 * @function
 * @param {object} props
 * @property {object} props.inputProps - Key name, validators, and onBlur
 * @property {object} props.baseValue - Start value of field?
 * @property {object} props.updateClientValue - Updates client state
 * @property {object} props.rowProps - `label`, `validRow`, `message`
 *
 * @returns Component
 */
const MonthlyCashFlowRow = function ({ inputProps, baseValue, updateClientValue, rowProps }) {

  inputProps = {
    ...inputProps, // name, validators, and onBlur
    className: 'cashflow-column',
    format:    toMoneyStr,
    store:     updateClientValue,
  };

  return (
    <CashFlowContainer { ...rowProps }>
      <ManagedNumberField
        { ...inputProps }
        value={ baseValue }
        otherData={{ interval: 'monthly' }} />
    </CashFlowContainer>
  );

};  // End <MonthlyCashFlowRow>


// Ideas of how to handle a different styling situation
// (if we swap the input and label positions)

// If we want more control over placement, we may look into this:
// <Grid textAlign='center' verticalAlign='middle'>
//   <Grid.Row className='inputs-in-right-column'>
//     <Grid.Column className='left-label'>
//       <label>Earned Income</label>
//     </Grid.Column>
//     <Grid.Column className='right-input'>
//       <Input type='number'/>
//     </Grid.Column>
//   </Grid.Row>
// </Grid>


export {
  CashFlowContainer,
  CashFlowInputsRow,
  CashFlowDisplayRow,
  MonthlyCashFlowRow,
};
