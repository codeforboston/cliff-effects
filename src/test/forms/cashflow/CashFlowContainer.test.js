import React from 'react';
import { mount } from 'enzyme';

import { CashFlowContainer } from '../../../forms/cashflow';

describe('<CashFlowContainer>', () => {
  const defaultProps = {
    label:    'Rent',
    children: <span>Some content</span>,
  };

  it('renders with no message', () => {
    const validProps = {
      ...defaultProps,
      validRow: true,
    };
    expect(mount(<CashFlowContainer { ...validProps } />)).toMatchSnapshot();
  });

  it('renders with message', () => {
    const invalidProps = {
      ...defaultProps,
      validRow: false,
      message:  'What did you enter!?',
    };
    expect(mount(<CashFlowContainer { ...invalidProps } />)).toMatchSnapshot();
  });
});
