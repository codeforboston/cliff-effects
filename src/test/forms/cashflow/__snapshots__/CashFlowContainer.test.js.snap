// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CashFlowContainer> renders with message 1`] = `
<CashFlowContainer
  label="Rent"
  message="What did you enter!?"
  validRow={false}
>
  <FormField
    className="cashflow"
    inline={true}
  >
    <div
      className="inline field cashflow"
    >
      <span>
        Some content
      </span>
      <div
        className="cashflow-column cashflow-column-last-child"
      >
        <label>
          Rent
        </label>
      </div>
      <InvalidMessage
        message="What did you enter!?"
        validRow={false}
      >
        <Label
          basic={true}
          color="red"
          pointing="left"
        >
          <div
            className="ui red left pointing basic label"
            onClick={[Function]}
          >
            What did you enter!?
          </div>
        </Label>
      </InvalidMessage>
    </div>
  </FormField>
</CashFlowContainer>
`;

exports[`<CashFlowContainer> renders with no message 1`] = `
<CashFlowContainer
  label="Rent"
  validRow={true}
>
  <FormField
    className="cashflow"
    inline={true}
  >
    <div
      className="inline field cashflow"
    >
      <span>
        Some content
      </span>
      <div
        className="cashflow-column cashflow-column-last-child"
      >
        <label>
          Rent
        </label>
      </div>
      <InvalidMessage
        validRow={true}
      />
    </div>
  </FormField>
</CashFlowContainer>
`;
